test/mock/README.md

```diff
@@ -0,0 +1,162 @@
+Mock
+----
+
+# Overview
+
+The origin/mock is a docker image that is built during 'make release'.
+Its main purpose is to have a flexible component that can safely be
+used and asserted in tests and that have an expected behaviour:
+
+- CRUD on key/value,
+- suicide,
+- simple redirect
+- chaining call concept 
+
+The server must have MOCK_NAME as env variable that will contains
+the name of the server otherwise it raise a panic.
+
+By default the server serves on the port 8080. By setting environment
+variable MOCK_PORT to another port the server will use this particular
+port.
+
+# Server spawning
+
+Although this server is expected to run as docker container you can
+run it this way:
+
+    $ cd test/mock
+    $ MOCK_NAME=mock1 go run server.go
+    Started mock1, serving at 8080
+
+Starting a second server:
+
+    $ MOCK_NAME=mock2 MOCK_PORT=8081 go run server.go
+    Started mock2, serving at 8081
+    
+# API
+
+## /name
+
+Returns the name of the server
+
+    $ curl http://localhost:8080/name
+    mock1
+    
+    $ curl http://localhost:8081/name
+    mock2
+
+## /put/{key}/{value}
+
+Puts the value "value" under the key {key}. Always returns ok.
+
+    $ curl http://localhost:8080/put/k1/value1
+    ok
+    $ curl http://localhost:8080/put/k2/value2
+    ok
+    $ curl http://localhost:8080/put/k3/value3
+    ok
+
+## /contains/{key}
+
+Returns "ok" if the server contains {key}, "ko" otherwise.
+
+    $ curl http://localhost:8080/contains/k1
+    ok
+     
+    $ curl http://localhost:8081/contains/k1
+    ko
+     
+## /get/{key}
+
+Returns the value behind the key {key}, empty string if there is no
+value for this key.
+
+    $ curl http://localhost:8080/get/k1
+    value1
+
+    $ curl http://localhost:8081/get/k1
+    ""           <-- empty string without "
+
+## /keys
+
+Returns all the keys
+
+   $ curl http://localhost:8080/keys
+   k1
+   k2
+   k3
+
+## /delete/{key}
+
+Deletes the entry where key {key} is stored. Returns "ok" if deletion occurs, "ko" if not.
+
+    $  curl http://localhost:8080/delete/k2
+    ok
+    
+    $  curl http://localhost:8080/delete/k2
+    ko
+    
+## /redirect/{scheme}/{host}/{port}/{path}
+
+This redirects the contents of the url {scheme}://{host}:{port}/{path} 
+
+    $ curl http://localhost:8080/redirect/http/localhost/8081/name
+    Querying http://localhost:8081/name got response mock2
+
+
+## /suicide
+
+This kills the server itself immediatly.
+
+## /suicide/{delay}
```

Comment:
I like this feature but I'm not sure about the name.  How about 'selfdestruct' instead?